### DISCUSSION TITLE
Is it possible to install only the types of Cypress?

### DISCUSSION THREAD

**Discussion by CharlieBrownCharacter** [cypress-io/cypress#22729] [Questions and Help]
I'm developing a plugin that will help create data on the backend of several frameworks and I would like to add a custom command `create` to Cypress. What I have so far is:

```typescript
export const createCommand = (model: string, options: FactoryOptions) => {
  ...
  return cy.request({ ... });
}
```

Then the user would import this function in his commands as:

_commands.js_
```typescript
import { createCommand } from '@codotto/cypress";

Cypress.Commands.add('createModel', createCommand);
```

What I would like is to be able to have typed in my `createCommand` without having to install Cypress completely in my package repository. Is this possible?

Besides the initial question, if you have any suggestion on how I could add this command to Cypress without the user having to add it manually, I would highly appreciate.

**Comment by portyu9** [comment]
Hi @CharlieBrownCharacter 

Cypress ships its TypeScript definitions inside the `cypress` package, so there is no separate `@types/cypress`.

For a Cypress plugin, declare Cypress as a peer dependency:

```json
{
  "peerDependencies": {
    "cypress": ">=10"
  }
}
```

For development/type checking, install the Cypress package without downloading its binary:

```bash
CYPRESS_INSTALL_BINARY=0 npm install --save-dev cypress
```

Then you can type your command normally:

```ts
export const createCommand: Cypress.CommandFn<'createModel'> = (...) => {
  return cy.request(...)
}
```

Your package can also export a setup module that registers the command; the consumer only needs to import that module once from their Cypress support file.

Please mark this answer as accepted if it helped.
